core/planner/planner.go

```diff
@@ -77,41 +76,51 @@ func (t *Planner) Plan(rates api.Rates, requiredDuration time.Duration, targetTi
 	return plan
 }
 
-// Active determines if current slot should be used for charging for a total required duration until target time
-func (t *Planner) Active(requiredDuration time.Duration, targetTime time.Time) (time.Time, time.Time, bool, error) {
+// Plan creates a lowest-cost charging plan, considering edge conditions
+func (t *Planner) Plan(requiredDuration time.Duration, targetTime time.Time) (api.Rates, error) {
 	if t == nil || requiredDuration <= 0 {
-		return time.Time{}, time.Time{}, false, nil
+		return nil, nil
 	}
 
 	// calculate start time
 	latestStart := targetTime.Add(-requiredDuration)
 	afterStart := t.clock.Now().After(latestStart) || t.clock.Now().Equal(latestStart)
 	beforeTarget := t.clock.Now().Before(targetTime)
 
+	var plan api.Rates
+	if afterStart && beforeTarget {
+		plan = api.Rates{
+			api.Rate{
+				Start: latestStart,
+				End:   targetTime,
+			},
+		}
+	}
+
 	// target charging without tariff or late start
 	if t.tariff == nil || afterStart {
-		return latestStart, targetTime, afterStart && beforeTarget, nil
+		return plan, nil
 	}
 
 	rates, err := t.tariff.Rates()
 
 	// treat like normal target charging if we don't have rates
 	if len(rates) == 0 || err != nil {
-		return latestStart, targetTime, afterStart && beforeTarget, err
+		return plan, err
 	}
 
 	// rates are by default sorted by date, oldest to newest
 	last := rates[len(rates)-1].End
 
 	// sort rates by price and time
-	sort.Sort(rates)
+	slices.SortStableFunc(rates, sortByCost)
 
 	// reduce planning horizon to available rates
 	if targetTime.After(last) {
 		// there is enough time for charging after end of current rates
 		durationAfterRates := targetTime.Sub(last)
 		if durationAfterRates >= requiredDuration {
-			return time.Time{}, time.Time{}, false, nil
+			return nil, nil
```

Comment:
@andig beim Testen laufe ich immer in diesen Fall. Direkt nach dem setzen einer Zielzeit wird die Projected Start Time erst korrekt angezeigt. In der zweiten Iteration lande ich aber hier und die Projected Start Time verschwindet.

Erster Lauf
```
[lp-1  ] DEBUG 2023/01/08 20:53:04 planned 3h50m17s until 2023-01-09 02:00:00 +0100 CET at 3680W: total plan duration: 50m17s, avg cost: 562.000
```

Zweiter Lauf:
```
[lp-1  ] DEBUG 2023/01/08 20:53:10 planned 3h50m17s until 2023-01-09 03:00:00 +0100 CET at 3680W: total plan duration: 0s, avg cost: NaN
```

Reply:
Hast du denn available rates? Also passt deine Tariftabelle?

Reply:
testest du mit Grünstromindex? Ich glaub da muss der Tarif noch die Rates kopieren bevor er sie zurück gibt! Sonst stimmt im zweiten durchlauf die Reihenfolge nicht mehr. Da muss der Tarif defensiver sein.

Reply:
Ja, ich habe mit Grünstromindex getestet (demo.yaml).